dependencies.yaml

```diff
@@ -281,13 +281,13 @@ dependencies:
       - output_types: conda
         packages:
           # Allow runtime version to float up to minor version
-          - libarrow-all==14.*
+          - libarrow-all>=14.0.1,<15.0.0a0
   pyarrow_run:
     common:
       - output_types: [conda, requirements, pyproject]
         packages:
           # Allow runtime version to float up to minor version
-          - pyarrow==14.*
+          - pyarrow>=14.0.1,<15.0.0a0
```

Comment:
Maybe worth noting here and above that in this particular case we're no longer allowing runtime versions to float down to 14.0.0 because of a CVE.

Reply:
```suggestion
          # Disallow pyarrow 14.0.0 due to a CVE
          - pyarrow>=14.0.1,<15.0.0a0
```